### PR TITLE
Resize cache

### PR DESCRIPTION
**Pull request by tobiasKaminsky**

### CODE REVIEW

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 1]
```diff
@@ -107,7 +109,19 @@ protected Void doInBackground(File... params) {
...
+                        SharedPreferences appPrefs =
+                                PreferenceManager.getDefaultSharedPreferences(MainApp.getAppContext());
+                        // due to backward compatibility
+                        Integer cacheSize = CACHE_SIZE_MB * 1024 * 1024;
+                        try {
+                            cacheSize = appPrefs.getInt("pref_cache_size", cacheSize);
+                        } catch (ClassCastException e) {
+                            String temp = appPrefs.getString("pref_cache_size",
+                                    cacheSize.toString());
+                            cacheSize = Integer.decode(temp) * 1024 * 1024;
```
Comment (przybylski) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: too bad that decode can throw ;)

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 2]
```diff
@@ -159,6 +173,31 @@ public static Bitmap getBitmapFromDiskCache(String key) {
...
+     * Sets max size of cache
+     * @param maxSize in MB
+     * @return
+     */
+    public static boolean setMaxSize(long maxSize){
+        if (mThumbnailCache != null){
+            mThumbnailCache.setMaxSize(maxSize * 1024 * 1024);
+            return true;
+        } else {
+            return false;
```
Comment (przybylski) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: why in else and not simply after `if` bracket?

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 3]
```diff
@@ -159,6 +173,31 @@ public static Bitmap getBitmapFromDiskCache(String key) {
...
+        }
+    }
+
+    /**
+     * Shows max cache size
+     * @return max cache size in MB.
+     */
+    public static long getMaxSize(){
+        if (mThumbnailCache == null) {
```
Comment (przybylski) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: as a presonal preference I think you should first check for expected case `mThumbnailCache != null` and than go to fallback behavior (getting data from cache manager)

File: src/com/owncloud/android/ui/activity/Preferences.java [hunk 4]
```diff
@@ -145,6 +146,27 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             });
         }
 
+        final Preference pCacheSize = findPreference("pref_cache_size");
+        if (pCacheSize != null){
+            final SharedPreferences appPrefs =
+                    PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+            Long cacheSize = ThumbnailsCacheManager.getMaxSize();
+            pCacheSize.setSummary(cacheSize + " Mb");
```
Comment (przybylski) [inline src/com/owncloud/android/ui/activity/Preferences.java]: Mb is suggesting that this is megaBIT is is right?

File: src/com/owncloud/android/ui/activity/Preferences.java [hunk 5]
```diff
@@ -145,6 +146,27 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
...
+            pCacheSize.setOnPreferenceChangeListener(new OnPreferenceChangeListener() {
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    int size = Integer.decode((String) newValue);
+                    if (ThumbnailsCacheManager.setMaxSize(size)){
+                        appPrefs.edit().putInt("pref_cache_size", size);
```
Comment (przybylski) [inline src/com/owncloud/android/ui/activity/Preferences.java]: Without commit/apply this makes no sense

**Comment by tobiasKaminsky** [comment]
Due to our recent discussion about a more intelligent mechanism for cache size I postpone this (and maybe it will never be merged)

**Comment by tobiasKaminsky** [comment]
As discussed in another issue this should be done in a real smart way by the app itself and not bother an inexperienced user.

**Pull request closed without merging**